src/runtime_src/core/common/api/xrt_device.cpp

```diff
@@ -289,28 +291,39 @@ get_info(info::device param) const
   case info::device::offline :
     return query::raw<info::device::offline, xrt_core::query::is_offline>(handle.get());
   case info::device::electrical :            // std::string
-    return query::json_str(xrt_core::sensor::read_electrical(handle.get()));
+    return query::json_str(xrt_core::sensor::read_electrical(handle.get()), abi);
   case info::device::thermal :               // std::string
-    return query::json_str(xrt_core::sensor::read_thermals(handle.get()));
+    return query::json_str(xrt_core::sensor::read_thermals(handle.get()), abi);
   case info::device::mechanical :            // std::string
-    return query::json_str(xrt_core::sensor::read_mechanical(handle.get()));
+    return query::json_str(xrt_core::sensor::read_mechanical(handle.get()), abi);
   case info::device::memory :                // std::string
-    return query::json_str(xrt_core::memory::memory_topology(handle.get()));
+    return query::json_str(xrt_core::memory::memory_topology(handle.get()), abi);
   case info::device::platform :              // std::string
-    return query::json_str(xrt_core::platform::platform_info(handle.get()));
+    return query::json_str(xrt_core::platform::platform_info(handle.get()), abi);
   case info::device::pcie_info :                  // std::string
-    return query::json_str(xrt_core::platform::pcie_info(handle.get()));
+    return query::json_str(xrt_core::platform::pcie_info(handle.get()), abi);
   case info::device::dynamic_regions :         // std::string
-    return query::json_str(xrt_core::memory::xclbin_info(handle.get()));
+    return query::json_str(xrt_core::memory::xclbin_info(handle.get()), abi);
   case info::device::host :                   // std::string
     boost::property_tree::ptree pt;
     xrt_core::get_xrt_build_info(pt);
-    return query::json_str(pt);
+    return query::json_str(pt, abi);
   }
 
   throw std::runtime_error("internal error: unreachable");
 }
 
+// Deprecated but left for support of old existing binaries in the
+// field that reference this symbol. Unused since xrt-2.12.x
+boost::any
+device::
+get_info(info::device param) const
```

Comment:
So, you deprecate this.
But you could also keep this by default and not advertising the one with the ABI, if the application does not care about the ABI besides just using the one it does understand, which means the default one defines in the headers when compiled.

Reply:
I don't advertise the ABI one.   The only public API is templated xrt::device::get_info(), which now calls implementation get_info(..., abi).     The ABI argument is an implementation detail.   The old get_info() implementation would be referenced in binaries compiled against xrt-2.11.x, so in order for these binaries to continue to run (without recompilation), the old get_info symbol is preserved.